core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -2042,6 +2057,108 @@ private long copyUnordered(RecordCursor cursor, TableWriterAPI writer, RecordToR
         return rowCount;
     }
 
+    private void createMatView(final ExecutionModel model, SqlExecutionContext executionContext) throws SqlException {
+        final CreateMatViewModel createMatViewModel = (CreateMatViewModel) model;
+        final ExpressionNode name = createMatViewModel.getName();
+
+        // Fast path for CREATE MATERIALIZED VIEW IF NOT EXISTS in scenario when the view already exists
+        final int status = executionContext.getTableStatus(path, name.token);
+        if (createMatViewModel.isIgnoreIfExists() && status == TableUtils.TABLE_EXISTS) {
+            final TableToken tt = engine.getTableTokenIfExists(name.token);
+            if (!tt.isMatView()) {
+                throw SqlException.$(name.position, "A table already exists with this name");
+            }
+            compiledQuery.ofCreateMatView(tt);
+        } else if (status == TableUtils.TABLE_EXISTS) {
+            throw SqlException.$(name.position, "A view or a table already exists with this name");
+        } else {
+            final TableToken matViewToken;
+            this.insertCount = -1;
+            if (createMatViewModel.getQueryModel() != null) {
+                matViewToken = createMatViewFromCursorExecutor(createMatViewModel, executionContext, name.position);
+                createMatViewModel.setTableToken(matViewToken);
+            } else {
+                throw SqlException.position(0).put("cannot create materialized view without a query");
+            }
+
+            final MaterializedViewDefinition matViewDefinition = createMatViewModel.generateDefinition();
+
+            // TODO: persist view definition, i.e. save mat view metadata
+            //  load them back on startup
+
+            final TableToken baseTableToken = engine.getTableTokenIfExists(matViewDefinition.getBaseTableName());
+            engine.getMaterializedViewGraph().upsertView(baseTableToken, matViewDefinition);
+
+            compiledQuery.ofCreateMatView(matViewToken);
+        }
+    }
+
+    private TableToken createMatViewFromCursorExecutor(
+            CreateMatViewModel model,
+            SqlExecutionContext executionContext,
+            int position
+    ) throws SqlException {
+        executionContext.setUseSimpleCircuitBreaker(true);
+        try (
+                final RecordCursorFactory factory = generate(model.getQueryModel(), executionContext);
+                final RecordCursor cursor = factory.getCursor(executionContext)
+        ) {
+            typeCast.clear();
+            final RecordMetadata metadata = factory.getMetadata();
+            validateMatViewModelAndCreateTypeCast(model, metadata, typeCast);
```

Comment:
Essentially, we create the table from the query metadata, the validation that column type matches is only confusing in this case. I removed this line and tests still passed.

Reply:
You are right, will change this.
Mat view model validation is needed, but create type cast syntax is not allowed for mat views, only for CREATE TABLE AS SELECT, so that will be removed.

Reply:
done